dandiapi/api/signals.py

```diff
@@ -0,0 +1,28 @@
+from allauth.account.signals import user_signed_up
+from corsheaders.signals import check_request_enabled
+from django.conf import settings
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from rest_framework.authtoken.models import Token
+
+from dandiapi.api.mail import send_registered_notice_email
+
+
+@receiver(check_request_enabled, dispatch_uid='cors_allow_anyone_read_only')
+def cors_allow_anyone_read_only(sender, request, **kwargs):
+    """Allow any read-only request from any origin."""
+    return request.method in ('GET', 'HEAD', 'OPTIONS')
```

Comment:
Does this cover our current need? If so, can we strip down our current list of allowed hosts in the Terraform setup? If not, we'll need to consult the list of hosts here to also grant access to the mutative operations if the request comes from one of those.

Reply:
This signal handler is basically a layer before the default CORS behavior. Before checking anything else, this will allow anyone to do any read-only operations. If a user is trying to do a POST/PUT/DELETE, this handler does not apply, so the server reverts to the default CORS behavior, which checks against the CORS_ORIGIN_WHITELIST.

@satra can better answer if there are any applications that are strictly read-only and can therefore be removed from CORS_ORIGIN_WHITELIST once this is merged.

Reply:
you could remove the codepen uris. we are not using it for anything other than `GET`.